packages/app/src/app/overmind/effects/vscode/sandboxFsSync/index.ts

```diff
@@ -0,0 +1,357 @@
+              fs: 'CodeSandboxEditorFS',
+              options: {
+                api: {
+                  getSandboxFs: options.getSandboxFs,
+                },
+              },
+            },
+            '/sandbox/node_modules': {
+              fs: 'CodeSandboxFS',
+              options: getTypeFetcher().options,
+            },
+            '/vscode': {
+              fs: 'LocalStorage',
+            },
+            '/home': {
+              fs: 'LocalStorage',
+            },
+            '/extensions': {
+              fs: 'BundledHTTPRequest',
+              options: {
+                index: EXTENSIONS_LOCATION + '/extensions/index.json',
+                baseUrl: EXTENSIONS_LOCATION + '/extensions',
+                bundle: EXTENSIONS_LOCATION + '/bundles/main.min.json',
+                logReads: process.env.NODE_ENV === 'development',
+              },
+            },
+            '/extensions/custom-theme': {
+              fs: 'InMemory',
+            },
+          },
+        },
+        async e => {
+          if (e) {
+            reject(e);
+          } else {
+            resolve();
+          }
+        }
+      );
+    });
+  }
+
+  public create(sandbox: Sandbox): SandboxFs {
+    const sandboxFs = {};
+
+    sandbox.modules.forEach(m => {
+      const path = getModulePath(sandbox.modules, sandbox.directories, m.id);
+      if (path) {
+        m.path = path;
+        sandboxFs[path] = {
+          ...m,
+          type: 'file',
+        };
+      }
+    });
+
+    sandbox.directories.forEach(d => {
+      const path = getDirectoryPath(sandbox.modules, sandbox.directories, d.id);
+
+      d.path = path;
+      // If this is a single directory with no children
+      if (!Object.keys(sandboxFs).some(p => dirname(p) === path)) {
+        sandboxFs[path] = { ...d, type: 'directory' };
+      }
+    });
+
+    this.sync();
+
+    return sandboxFs;
+  }
+
+  public writeFile(fs: SandboxFs, module: Module) {
+    const copy = json(module);
+
+    writeFile(fs, copy);
+    this.send('write-file', copy);
+
+    if (module.title === 'package.json') {
+      this.syncDependencyTypings();
+    }
+  }
+
+  public rename(fs: SandboxFs, fromPath: string, toPath: string) {
+    rename(fs, fromPath, toPath);
+    this.send('rename', {
+      fromPath,
+      toPath,
+    });
+  }
+
+  public rmdir(fs: SandboxFs, directory: Directory) {
+    const copy = json(directory);
+
+    rmdir(fs, copy);
+    this.send('rmdir', copy);
+  }
+
+  public unlink(fs: SandboxFs, module: Module) {
+    const copy = json(module);
+
+    unlink(fs, copy);
+    this.send('unlink', copy);
+  }
+
+  public mkdir(fs: SandboxFs, directory: Directory) {
+    const copy = json(directory);
+
+    mkdir(fs, copy);
+    this.send('mkdir', copy);
+  }
+
+  private syncSandbox() {
+    // eslint-disable-next-line
+    console.log('## SYNCING SANDBOX AND TYPINGS WITH WORKERS');
+    this.send('sandbox-fs', json(this.options.getSandboxFs()));
```

Comment:
Maybe we should rename this to something different. It's not really the sandbox fs (I expect to get an interface with `readFile` etc), but rather a representation. Maybe we should rename this to something like `sandboxFiles`?